### PR TITLE
Hosted setup for VM testnets

### LINKED ISSUE #1184: Hosted node setup on VM testnets
Notes for implementation

- [x] Needs an HTTPS load balancer https://cloud.google.com/load-balancing/docs/https/

- [x] Get SSL certificate via https://www.terraform.io/docs/providers/acme/r/certificate.html, requires moving over to google DNS

- [x] Create DNS RR for {env}-infura.{domain}

Do we keep an internal lb for blockscout? Initial feeling is yes

### PR DESCRIPTION
**Pull request by tkporter**
### Description

Creates an hosted setup for VM testnets by using an external HTTPS load balancer. An SSL certificate is gotten from Let's Encrypt.

An internal & external load balancer can't use the same set of instances, so I now have 1 tx-node that is behind an internal load balancer to be used by internal services (blockscout, transaction-metrics exporter). It's probably worth it in the future to move this tx-node to just live in Kubernetes.

### Tested

Ran own testnet called `trevor4` & was able to connect to https://trevor4-infura.celo-testnet.org

### Other changes

Fail if running a `testnet` command when the env is intended to be for VMs

### Related issues

- Fixes #1184 

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/celotool/src/lib/vm-testnet-utils.ts [hunk 1]
```diff
@@ -305,14 +305,17 @@ function getTerraformBackendConfigVars(celoEnv: string, terraformModule: string)
...
     ...getEnvVarValues(testnetEnvVars),
-    ethstats_host: `${celoEnv}-ethstats.${fetchEnv(envVar.CLUSTER_DOMAIN_NAME)}.org`,
+    dns_zone_name: dnsZoneName(domainName),
+    ethstats_host: `${celoEnv}-ethstats.${domainName}.org`,
     gcloud_secrets_bucket: secretsBucketName,
     gcloud_secrets_base_path: secretsBasePath(celoEnv),
     // only able to view objects (for accessing secrets)
     gcloud_vm_service_account_email: 'terraform-testnet@celo-testnet.iam.gserviceaccount.com',
     genesis_content_base64: genesisBuffer.toString('base64'),
+    infura_setup_host: `${celoEnv}-infura.${domainName}.org`,
```
Comment (timmoreton) [inline packages/celotool/src/lib/vm-testnet-utils.ts]: need to rename this

File: packages/terraform-modules/testnet/main.tf [hunk 2]
```diff
@@ -107,9 +111,13 @@ module "tx_node" {
...
   source = "./modules/tx-node-load-balancer"
   # variables
-  celo_env           = var.celo_env
-  network_name       = data.google_compute_network.network.name
-  tx_node_self_links = module.tx_node.self_links
+  celo_env                = var.celo_env
+  dns_zone_name           = var.dns_zone_name
+  gcloud_credentials_path = var.gcloud_credentials_path
+  gcloud_project          = var.gcloud_project
+  infura_setup_host       = var.infura_setup_host
```
Comment (timmoreton) [inline packages/terraform-modules/testnet/main.tf]: Need to update names here too

File: packages/terraform-modules/testnet/modules/tx-node-load-balancer/main.tf [hunk 3]
```diff
@@ -2,53 +2,195 @@ locals {
...
+
+resource "random_id" "internal" {
+  byte_length = 8
+}
+
+# We want to maintain websockets (which are not supposed by the HTTPS external
+# load balancer) & avoid unnecessary egress costs.
+# An internal & external load balancer cannot use the same instance group. To
+# get around this, we allocate 1 of the tx-nodes to be for internal load balancing.
+# It's still included in `static_nodes.json`, but not included in the infura-like
```
Comment (timmoreton) [inline packages/terraform-modules/testnet/modules/tx-node-load-balancer/main.tf]: ditto

File: packages/terraform-modules/testnet/modules/tx-node/startup.sh [hunk 4]
```diff
@@ -50,45 +50,54 @@ IN_MEMORY_DISCOVERY_TABLE_FLAG=""
...
+      --wsapi=eth,net,web3,debug \
+      --nodekey=$DATA_DIR/pkey \
+      --etherbase=$ACCOUNT_ADDRESS \
+      --networkid=${network_id} \
+      --syncmode=full \
+      --miner.verificationpool=${verification_pool_url} \
+      --consoleformat=json \
+      --consoleoutput=stdout \
+      --verbosity=${geth_verbosity} \
+      --ethstats=${tx_node_name}:$ETHSTATS_WEBSOCKETSECRET@${ethstats_host} \
```
Comment (timmoreton) [inline packages/terraform-modules/testnet/modules/tx-node/startup.sh]: chat to Dimi about this?

**Review by timmoreton — APPROVED** [review]

**Comment by tkporter** [comment]
Closing in favor of #2073, which included this PR

**Pull request closed without merging**